src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c

```diff
@@ -121,187 +120,241 @@ void EnSyatekiOkuta_Destroy(Actor* thisx, PlayState* play) {
-    this->actionFunc = func_80A36488;
+void EnSyatekiOkuta_SetupFloat(EnSyatekiOkuta* this) {
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, SG_OCTO_ANIM_FLOAT);
+    this->actionFunc = EnSyatekiOkuta_Float;
 }
 
-void func_80A36488(EnSyatekiOkuta* this, PlayState* play) {
+/**
+ * Floats in place until the Shooting Gallery Man tells it to hide.
+ */
+void EnSyatekiOkuta_Float(EnSyatekiOkuta* this, PlayState* play) {
     EnSyatekiMan* syatekiMan = (EnSyatekiMan*)this->actor.parent;
 
+    // In practice, if the Octorok is floating, then the octorokState is either SG_OCTO_STATE_APPEARED or
+    // SG_OCTO_STATE_HIDING. Only the latter state is greater than SG_OCTO_STATE_INITIAL, so that's what
+    // this check is looking for.
     if (syatekiMan->perGameVar1.octorokState >= SG_OCTO_STATE_INITIAL) {
-        func_80A364C0(this);
+        EnSyatekiOkuta_SetupHide(this);
     }
 }
 
-void func_80A364C0(EnSyatekiOkuta* this) {
-    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 2);
-    this->actionFunc = func_80A36504;
+void EnSyatekiOkuta_SetupHide(EnSyatekiOkuta* this) {
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, SG_OCTO_ANIM_HIDE);
+    this->actionFunc = EnSyatekiOkuta_Hide;
 }
 
-void func_80A36504(EnSyatekiOkuta* this, PlayState* play) {
+/**
+ * Retreats underwater, then makes the Octorok do nothing until the Shooting Gallery Man tells it to appear again.
+ */
+void EnSyatekiOkuta_Hide(EnSyatekiOkuta* this, PlayState* play) {
     if (Animation_OnFrame(&this->skelAnime, 4.0f)) {
-        func_80A361B0(this, play);
+        EnSyatekiOkuta_SpawnSplash(this, play);
         Actor_PlaySfx(&this->actor, NA_SE_EN_DAIOCTA_LAND);
     } else if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
-        func_80A362F8(this);
+        EnSyatekiOkuta_SetupDoNothing(this);
     }
 }
 
-void func_80A3657C(EnSyatekiOkuta* this) {
-    this->unk_2A4 = 0;
-    this->unk_2AA = 300;
-    if (this->unk_2A6 == 1) {
+void EnSyatekiOkuta_SetupDie(EnSyatekiOkuta* this) {
+    this->deathTimer = 0;
+    this->hitResultAlpha = 300;
+    if (this->type == SG_OCTO_TYPE_RED) {
         Actor_PlaySfx(&this->actor, NA_SE_EN_OCTAROCK_DEAD1);
     }
 
-    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, 1);
-    this->actionFunc = func_80A365EC;
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, SG_OCTO_ANIM_DIE);
+    this->actionFunc = EnSyatekiOkuta_Die;
 }
 
-void func_80A365EC(EnSyatekiOkuta* this, PlayState* play) {
-    Vec3f sp84;
-    Vec3f sp78;
+/**
+ * Plays the death animation and slowly shrinks the Octorok. Also spawns various bubble and dust
+ * effects as it dies. Once the Octorok is finished with its death animation, this function will
+ * make it do nothing until the Shooting Gallery Man tells it to appear again.
+ */
+void EnSyatekiOkuta_Die(EnSyatekiOkuta* this, PlayState* play) {
```

Comment:
So the `Octorok` doesn't die in the sense that it's despawned, but more goes into hiding until it can reappear for the next round it's involved in?

Reply:
Yeah, that's correct. You'll notice that there's only a single `Actor_Kill` in here, and it's in the Init function. In other words, once the Octorok is spawned, it *stays* spawned forever; "killing" it just makes it hide until it's ready again.